### ISSUE TITLE
Add option to color-named rule to prevent bad color names

### ISSUE THREAD

**Issue by halis** [stylelint/stylelint#2418]
> Describe the issue. Is it a bug or a feature request (new rule, new option, etc.)?

New option

> Which rule, if any, is this issue related to?

`color-named`

> What CSS is needed to reproduce this issue?

```css
.foo {
  color: greem;
}
```

> What stylelint configuration is needed to reproduce this issue?

I am proposing this syntax:
```json
{
  "rules": {
    "color-named": "no-unknown-names"
  }
}
```
Which would in turn flag the CSS because `greem` is an unknown name.

**Comment by halis** [comment]
The data to make this happen is in the repo:
https://github.com/stylelint/stylelint/blob/master/lib/reference/namedColorData.js

**Comment by evilebottnawi** [comment]
@halis i think best name is `color-named-no-unknown`

**Comment by halis** [comment]
Seems fine. 

To make sure I explained it correctly above,
I would expect this option to:
1) Allow colors in hex/rgba
2) Allow color names
3) Only trigger an error if the color name is not in the list of colors

**Comment by evilebottnawi** [comment]
@stylelint/core what do you think about rule?

**Comment by CAYdenberg** [comment]
Wait, so is the proposal for a new rule, or a new option for the existing rule? What if you want no unknown names, but also to use names wherever possible?

**Comment by davidtheclark** [comment]
@evilebottnawi Seems fine to me.

@CAYdenberg "use names wherever possible" would be a separate rule.

**Comment by jeddy3** [comment]
> Wait, so is the proposal for a new rule, or a new option for the existing rule

A new rule. However, I believe this spec validation can already be done with [css-tree](https://csstree.github.io/docs/validator.html) via [`stylelint-csstree-validator`](https://github.com/csstree/stylelint-validator). Do we really want to replicate this in core?

> @CAYdenberg "use names wherever possible" would be a separate rule.

And _is_ a separate rule. This _perference_ can be enforced with the [`"always-where-possible"`](https://github.com/stylelint/stylelint/blob/master/lib/rules/color-named/README.md#always-where-possible) option of the [`color-named`](https://github.com/stylelint/stylelint/blob/master/lib/rules/color-named) rule.

**Comment by jeddy3** [comment]
Closed by https://github.com/stylelint/stylelint/pull/2704
